### PR TITLE
feat(locale-modal): add shadow parts

### PR DESCRIPTION
**Pull request by marcelojcs**
### Related Ticket(s)

[JIRA](https://jsw.ibm.com/browse/ADCMS-5167)

### Description

{All non-slot elements in the shadow DOM should be given a unique "part" name allowing CSS to target and override component default styles.
R is changing in the application}}

### Changelog

**New**

Added Shadow Parts to the locale-modal component

### CODE REVIEW

**Review by andy-blum — CHANGES_REQUESTED** [review]
This needs work still. Try searching the `locale-modal` directory for instances of "html`" There are some spots that have been missed. Also, the "-composite" component renders in the light-dom so we don't need to add parts to anything it renders.

**Comment by marcelojcs** [comment]
@andy-blum 
Changes Addressed in my latest commit

File: packages/web-components/src/components/locale-modal/locale-search.ts [hunk 1]
```diff
@@ -217,14 +222,17 @@ class C4DLocaleSearch extends ThrottedInputMixin(
             size="${INPUT_SIZE.LARGE}"
             data-autoid="${c4dPrefix}--locale-modal__filter">
           </cds-search>
-          <div class="${prefix}--visually-hidden" aria-live="polite"></div>
-          <p class="${c4dPrefix}--locale-modal__search-text">
+          <div
+            class="${prefix}--visually-hidden"
+            aria-live="polite"
+            part="polite"></div>
+          <p class="${c4dPrefix}--locale-modal__search-text" part="seach-text">
```
Comment (m4olivei) [inline packages/web-components/src/components/locale-modal/locale-search.ts:229]:
```suggestion
          <p class="${c4dPrefix}--locale-modal__search-text" part="search-text">
```

File: packages/web-components/src/components/locale-modal/locale-search.ts [hunk 2]
```diff
@@ -217,14 +222,17 @@ class C4DLocaleSearch extends ThrottedInputMixin(
             size="${INPUT_SIZE.LARGE}"
             data-autoid="${c4dPrefix}--locale-modal__filter">
           </cds-search>
-          <div class="${prefix}--visually-hidden" aria-live="polite"></div>
-          <p class="${c4dPrefix}--locale-modal__search-text">
+          <div
+            class="${prefix}--visually-hidden"
+            aria-live="polite"
+            part="polite"></div>
```
Comment (m4olivei) [inline packages/web-components/src/components/locale-modal/locale-search.ts:228]: `polite` refers to how [`aria-live` updates](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) behave for users of assistive tech, it doesn't really describe this element. I want to suggest something like `aria-updates`
```suggestion
            part="aria-updates"></div>
```

File: packages/web-components/src/components/locale-modal/region-item.ts [hunk 3]
```diff
@@ -49,8 +55,15 @@ class C4DRegionItem extends C4DCard {
...
     const { _classes: classes, _cardClasses: cardClasses } = this;
     return html`
-      <button id="link" class="${classes}" disabled type="button">
-        <div class="${cardClasses}">${this._renderInner()}</div>
+      <button
+        id="link"
+        class="${classes}"
+        disabled
+        type="button"
+        part="button disabled-button">
```
Comment (m4olivei) [inline packages/web-components/src/components/locale-modal/region-item.ts:63]: Lets follow a convention we've been using elsewhere:
```suggestion
        part="button button--disabled">
```

File: packages/web-components/src/components/locale-modal/region-item.ts [hunk 4]
```diff
@@ -60,8 +73,14 @@ class C4DRegionItem extends C4DCard {
     return disabled
       ? this._renderDisabledLink()
       : html`
-          <button id="link" class="${classes}" type="button">
-            <div class="${cardClasses}">${this._renderInner()}</div>
+          <button
+            id="link"
+            class="${classes}"
+            type="button"
+            part="button enabled-button">
```
Comment (m4olivei) [inline packages/web-components/src/components/locale-modal/region-item.ts:80]: Lets follow a convention we've been using elsewhere:
```suggestion
            part="button button--enabled">
```

**Review by m4olivei — APPROVED** [review]
Thanks for the changes! All set here.

**Pull request merged**